BUILD.bazel

```diff
@@ -36,1458 +38,102 @@ gz_configure_header(
-        ":Quaternion",
-    ],
-)
-
-cc_test(
-    name = "Cylinder_TEST",
-    srcs = ["src/Cylinder_TEST.cc"],
-    deps = [
-        ":Cylinder",
-        "@googletest//:gtest",
-        "@googletest//:gtest_main",
-    ],
-)
-
-cc_library(
-    name = "DiffDriveOdometry",
-    srcs = ["src/DiffDriveOdometry.cc"],
-    hdrs = [
-        "include/gz/math/DiffDriveOdometry.hh",
-    ],
-    includes = ["include"],
-    deps = [
-        ":Angle",
-        ":Config",
-        ":Export",
-        ":RollingMean",
-        "@gz-utils//:ImplPtr",
-    ],
-)
-
-cc_test(
-    name = "DiffDriveOdometry_TEST",
-    srcs = ["src/DiffDriveOdometry_TEST.cc"],
-    deps = [
-        ":Angle",
-        ":DiffDriveOdometry",
-        ":Helpers",
-        "@googletest//:gtest",
-        "@googletest//:gtest_main",
+public_headers = glob(
+    include = [
+        "include/gz/math/*.hh",
+        "include/gz/math/detail/*.hh",
+        "include/gz/math/graph/*.hh",
     ],
-)
-
-cc_library(
-    name = "Ellipsoid",
-    hdrs = [
-        "include/gz/math/Ellipsoid.hh",
-        "include/gz/math/detail/Ellipsoid.hh",
-    ],
-    includes = ["include"],
-    deps = [
-        ":Helpers",
-        ":Inertial",
-        ":MassMatrix3",
-        ":Material",
-    ],
-)
-
-cc_library(
-    name = "Error",
-    srcs = ["src/Error.cc"],
-    hdrs = ["include/gz/math/detail/Error.hh"],
-    includes = ["include"],
-    deps = [
-        ":Config",
-        ":Export",
-    ],
-)
+) + [
+    "include/gz/math/Export.hh",
+    "include/gz/math/config.hh",
+]
 
-cc_test(
-    name = "Ellipsoid_TEST",
-    srcs = ["src/Ellipsoid_TEST.cc"],
-    deps = [
-        ":Ellipsoid",
-        ":Helpers",
-        ":Vector3",
-        "@googletest//:gtest",
-        "@googletest//:gtest_main",
+private_headers = glob(
+    include = [
+        "src/*.hh",
     ],
 )
 
-cc_library(
-    name = "Filter",
-    hdrs = ["include/gz/math/Filter.hh"],
-    includes = ["include"],
-    deps = [
-        ":Config",
-        ":Helpers",
-        ":Quaternion",
-        ":Vector3",
+sources = glob(
+    include = [
+        "src/*.cc",
     ],
-)
-
-cc_test(
-    name = "Filter_TEST",
-    srcs = ["src/Filter_TEST.cc"],
-    deps = [
-        ":Filter",
-        "@googletest//:gtest",
-        "@googletest//:gtest_main",
+    exclude = [
+        "src/*_TEST.cc",
     ],
 )
 
 cc_library(
-    name = "Frustum",
-    srcs = ["src/Frustum.cc"],
-    hdrs = ["include/gz/math/Frustum.hh"],
-    includes = ["include"],
-    deps = [
-        ":Angle",
-        ":AxisAlignedBox",
-        ":Config",
-        ":Matrix4",
-        ":Plane",
-        ":Pose3",
-        "@gz-utils//:ImplPtr",
-    ],
-)
-
-cc_test(
-    name = "Frustum_TEST",
-    srcs = ["src/Frustum_TEST.cc"],
-    deps = [
-        ":Frustum",
-        ":Helpers",
-        "@googletest//:gtest",
-        "@googletest//:gtest_main",
+    name = "gz-math",
+    srcs = sources,
+    hdrs = public_headers + private_headers,
```

Comment:
Private headers usually go in `srcs` instead.

> For cc_library rules, headers in hdrs comprise the public interface of the library and can be directly included both from the files in hdrs and srcs of the library itself as well as from files in hdrs and srcs of cc_* rules that list the library in their deps. Headers in srcs must only be directly included from the files in hdrs and srcs of the library itself. When deciding whether to put a header into hdrs or srcs, you should ask whether you want consumers of this library to be able to directly include it. This is roughly the same decision as between public and private visibility in programming languages.

https://bazel.build/reference/be/c-cpp#hdrs